modules/stochastic_tools/include/utils/AdaptiveMonteCarloUtils.h

```diff
@@ -78,4 +79,25 @@ Real computeMin(const std::vector<Real> & data);
  */
 std::vector<Real> computeVectorABS(const std::vector<Real> & data);
 
+/**
+ * propose a new sample using ComponentWise-MH.
+ *
+ * @param the current sample
+ * @param the random seed
+ */
+Real proposeNewSampleComponentWiseMH(const Real x, const Real rnd1, const Real rnd2);
+
+/**
+ * propose a new sample w/ regular MH. (AdaptiveImportanceSampler)
+ *
+ * @param the input distributions
+ * @param the random seed
+ * @param the input vector from the reporter
+ * @param the previously accepted samples
+ */
+std::vector<Real> proposeNewSampleMH(std::vector<const Distribution *> distributions,
+                                     const Real rnd,
+                                     const std::vector<std::vector<Real>> inputs,
+                                     std::vector<std::vector<Real>> _inputs_sto);
```

Comment:
Hi @zachmprince , I just wonder why "&" might be necessary here in the declaration. I tried adding "&"s before "distribution", "inputs" and "inputs_sto", and I always end up with this error:
```
dyld[24799]: missing symbol called
Abort trap: 6
```
Adding "const" works though and seems reasonable to me. Does it look alright to you if I don't go with the suggestion of adding "&"s? Thanks

Reply:
That does not make sense. You added the `&` in both the header and source file? The `&` is really important because it avoids copying the vectors when calling the function.

Reply:
You are right. I added them again and now everything works. Thanks for pointing that out!